AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/MutationSync/OutgoingMutationQueue/OutgoingMutationQueue.swift

```diff
@@ -179,36 +180,51 @@ final class OutgoingMutationQueue: OutgoingMutationQueueBehavior {
-                self.completeProcessingEvent(mutationEvent)
-                return
+    private func processSyncMutationToCloudResult(_ result: GraphQLOperation<MutationSync<AnyModel>>.Event,
+                                                  mutationEvent: MutationEvent,
+                                                  api: APICategoryGraphQLBehavior) {
+        if case let .completed(response) = result, case let .failure(graphQLResponseError) = response {
+            let processMutationErrorFromCloudOperation = ProcessMutationErrorFromCloudOperation(
+                dataStoreConfiguration: self.dataStoreConfiguration,
+                mutationEvent: mutationEvent,
+                api: api,
+                storageAdapter: self.storageAdapter,
+                graphQLResponseError: graphQLResponseError) { result in
+                    if case let .success(mutationEventOptional) = result,
+                        let outgoingMutationEvent = mutationEventOptional {
+                        self.outgoingMutationQueueSubject.send(outgoingMutationEvent)
+                    }
+                    self.completeProcessingEvent(mutationEvent)
             }
-
-            let processMutationErrorFromCloudOperation =
-                ProcessMutationErrorFromCloudOperation(dataStoreConfiguration: self.dataStoreConfiguration,
-                                                       mutationEvent: mutationEvent,
-                                                       api: api,
-                                                       storageAdapter: self.storageAdapter,
-                                                       graphQLResponseError: graphQLResponseError) { result in
-
-                self.log.verbose("[ProcessMutationErrorFromCloudOperation] mutation error processed, result: \(result)")
-                if case let .success(mutationEventOptional) = result,
-                    let outgoingMutationEvent = mutationEventOptional {
-                    self.outgoingMutationQueueSubject.send(outgoingMutationEvent)
-                }
-                self.completeProcessingEvent(mutationEvent)
+            self.operationQueue.addOperation(processMutationErrorFromCloudOperation)
+        } else if case let .failed(apiError) = result {
+            let processMutationErrorFromCloudOperation = ProcessMutationErrorFromCloudOperation(
+                dataStoreConfiguration: self.dataStoreConfiguration,
+                mutationEvent: mutationEvent,
+                api: api,
+                storageAdapter: self.storageAdapter,
+                apiError: apiError) { result in
+                    if case let .success(mutationEventOptional) = result,
+                        let outgoingMutationEvent = mutationEventOptional {
+                        self.outgoingMutationQueueSubject.send(outgoingMutationEvent)
+                    }
+                    self.completeProcessingEvent(mutationEvent)
```

Comment:
rather than repeat this twice, maybe it would be better to have:
```
    private func processSyncMutationToCloudResult(_ result: GraphQLOperation<MutationSync<AnyModel>>.Event,
                                                  mutationEvent: MutationEvent,
                                                  api: APICategoryGraphQLBehavior) {
        if case let .completed(response) = result, case let .failure(graphQLResponseError) = response {
            processMutationErrorFromCloudOperation(mutationEvent: mutationEvent,
                                                   api: api,
                                                   apiError: nil,
                                                   graphQLResponseError: graphQLResponseError)
        } else if case let .failed(apiError) = result {
            processMutationErrorFromCloudOperation(mutationEvent: mutationEvent,
                                                   api: api,
                                                   apiError: apiError,
                                                   graphQLResponseError: nil)

        } else {
            completeProcessingEvent(mutationEvent)
        }
    }

    func processMutationErrorFromCloudOperation(mutationEvent: MutationEvent,
                                                api: APICategoryGraphQLBehavior,
                                                apiError: APIError?,
                                                graphQLResponseError: GraphQLResponseError<MutationSync<AnyModel>>?) {
        let processMutationErrorFromCloudOperation = ProcessMutationErrorFromCloudOperation(
            dataStoreConfiguration: dataStoreConfiguration,
            mutationEvent: mutationEvent,
            api: api,
            storageAdapter: storageAdapter,
            graphQLResponseError: graphQLResponseError,
            apiError: apiError) { result in
                if case let .success(mutationEventOptional) = result,
                    let outgoingMutationEvent = mutationEventOptional {
                    self.outgoingMutationQueueSubject.send(outgoingMutationEvent)
                }
                self.completeProcessingEvent(mutationEvent)
        }
        operationQueue.addOperation(processMutationErrorFromCloudOperation)
    }
```

Reply:
thanks for the code snippet!